scripts/validate_changelog_utils.py

```diff
@@ -0,0 +1,108 @@
+import re
+
+changelog_diff_regex = "^([\s]*)diff --git a(.*)\/CHANGELOG.md b(.*)\/CHANGELOG.md"
+changelog_filename = "CHANGELOG.md"
+any_diff_substring = "diff --git"
+diff_file_start_regex = "^([\s]*)\@\@(.*)\@\@"
+pr_link_regex = "\[#\d+]\(https:\/\/github\.com\/mapbox\/mapbox-navigation-android\/pull\/\d+\)"
+
+def is_line_added(line):
+    return line.strip().startswith('+')
+
+def remove_plus(line):
+    return line.strip()[1:]
+
+def group_by_versions(lines):
+    groups = {}
+    group = []
+    group_name = ""
+    for line in lines:
+        if line.startswith("##") and len(line) > 2 and line[2] != '#':
+            if (len(group) > 0):
+                if len(group_name.strip()) > 0:
+                    groups[group_name] = group
+                group = []
+            group_name = line
+        elif len(line.strip()) > 0:
+            group.append(line)
+    if len(group) > 0 and len(group_name.strip()) > 0:
+        groups[group_name] = group
+    return groups
+
+def extract_unreleased_group(versions):
+    for version in versions.keys():
+        if 'Unreleased' in version:
+            return versions[version]
+    raise Exception("No 'Unreleased' section in CHANGELOG")
+
+def extract_stable_versions(versions):
+    str_before_version = "Mapbox Navigation SDK "
+    str_after_version = " "
+    stable_versions = {}
+    pattern = re.compile("[0-9]+\.[0-9]+\.[0-9]+")
+    for version in versions.keys():
+        beginIndex = version.find(str_before_version)
+        if beginIndex == -1:
+            continue
+        version_name = version[(beginIndex + len(str_before_version)):]
+        endIndex = version_name.find(str_after_version)
+        if endIndex == -1:
+            continue
+        version_name = version_name[:endIndex]
+        if pattern.fullmatch(version_name) != None:
+            stable_versions[version_name] = versions[version]
+    return stable_versions
+
+def should_skip_changelog(response_json):
+    if "labels" in response_json:
+        pr_labels = response_json["labels"]
+        for label in pr_labels:
+            if label["name"] == "skip changelog":
+                return True
+    return False
```

Comment:
maybe we shouldn't skip changelog checks if changelog file is modified?

Reply:
I think historically this label means "skip changelog checks". Anyways, I think it should mean this now. Maybe we should just rename it?
And we should skip the checks if it was modified and the label is present. Because there are more checks now and we can ignore, as you described, fixing a typo or clarifying an entry. For the simplicity of it I'd like to keep one label only.